File: app/src/main/java/com/fthdgn/books/dependecy/module/ApiModule.java
package com.fthdgn.books.dependecy.module;

import android.content.SharedPreferences;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import com.fthdgn.books.api.AccountManager;

@Module
public class ApiModule {

    @Provides
    @Singleton
    AccountManager providesAccountManager(SharedPreferences sharedPreferences) {
        return new AccountManager(sharedPreferences);
    }

}

File: app/src/main/java/com/fthdgn/books/dependecy/module/ApiModule.kt
package com.fthdgn.books.dependecy.module

import android.content.SharedPreferences
import com.fthdgn.books.api.AccountManager
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class ApiModule {

    @Provides
    @Singleton
    internal fun providesAccountManager(sharedPreferences: SharedPreferences): AccountManager =
            AccountManager(sharedPreferences)

}
